src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -53,20 +53,20 @@ public static class PSVersionInfo
         /// For each later release of PowerShell, this constant needs to
         /// be updated to reflect the right version.
         /// </remarks>
-        private static readonly Version s_psV1Version = new Version(1, 0);
-        private static readonly Version s_psV2Version = new Version(2, 0);
-        private static readonly Version s_psV3Version = new Version(3, 0);
-        private static readonly Version s_psV4Version = new Version(4, 0);
-        private static readonly Version s_psV5Version = new Version(5, 0);
-        private static readonly Version s_psV51Version = new Version(5, 1, NTVerpVars.PRODUCTBUILD, NTVerpVars.PRODUCTBUILD_QFE);
-        private static readonly SemanticVersion s_psV6Version = new SemanticVersion(6, 0, 0, preReleaseLabel: null, buildLabel: null);
-        private static readonly SemanticVersion s_psV61Version = new SemanticVersion(6, 1, 0, preReleaseLabel: null, buildLabel: null);
-        private static readonly SemanticVersion s_psV62Version = new SemanticVersion(6, 2, 0, preReleaseLabel: null, buildLabel: null);
-        private static readonly SemanticVersion s_psV7Version = new SemanticVersion(7, 0, 0, preReleaseLabel: null, buildLabel: null);
-        private static readonly SemanticVersion s_psV71Version = new SemanticVersion(7, 1, 0, preReleaseLabel: null, buildLabel: null);
-        private static readonly SemanticVersion s_psV72Version = new SemanticVersion(7, 2, 0, preReleaseLabel: null, buildLabel: null);
-        private static readonly SemanticVersion s_psSemVersion;
+        private static readonly Version s_psV1Version = new(1, 0);
+        private static readonly Version s_psV2Version = new(2, 0);
+        private static readonly Version s_psV3Version = new(3, 0);
+        private static readonly Version s_psV4Version = new(4, 0);
+        private static readonly Version s_psV5Version = new(5, 0);
+        private static readonly Version s_psV51Version = new(5, 1, NTVerpVars.PRODUCTBUILD, NTVerpVars.PRODUCTBUILD_QFE);
+        private static readonly Version s_psV6Version = new(6, 0, 0);
```

Comment:
I think we don't get perf win.
My preference to keep SemanticVersion.

Reply:
There is certainly a tiny-tiny perf win, because `PSCompatibleVersionsName` is a `Version[]` array. So, all `SemanticVersion` objects will be cast to `Version` objects when creating the array.

The casting for a `SemanticVersion` that doesn't have `label` and `build` part is to simply creating a `Version` object using `major, minor, and patch`, so by creating `Version` objects directly, the type casting is gone.